src/features/registry/registry.ts

```diff
@@ -207,13 +216,124 @@ const resolveImageId = multiCacheMemoizee(
 	},
 );
 
+const resolveImageLocation = multiCacheMemoizee(
+	async (
+		application: string,
+		release: string | undefined,
+		service: string | undefined,
+		tx: Tx,
+	): Promise<string | undefined> => {
+		try {
+			const [image] = await api.resin.get({
+				resource: 'image',
+				passthrough: { req: permissions.root, tx },
+				options: {
+					$top: 1,
+					$select: ['is_stored_at__image_location'],
+					$filter: {
+						release_image: {
+							$any: {
+								$alias: 'ri',
+								$expr: {
+									ri: {
+										is_part_of__release: {
+											$any: {
+												$alias: 'ipor',
+												$expr: {
+													ipor: {
+														status: 'success' as const,
+														belongs_to__application: {
+															$any: {
+																$alias: 'bta',
+																$expr: {
+																	bta: {
+																		slug: application,
+																	},
+																},
+															},
+														},
+														...(release == null && {
+															should_be_running_on__application: {
+																$any: {
+																	$alias: 'sbroa',
+																	$expr: {
+																		sbroa: {
+																			slug: application,
+																		},
+																	},
+																},
+															},
+														}),
+													},
+													...(release != null && {
+														$or: [
+															{ ipor: { commit: release } },
+															{
+																ipor: {
+																	semver: release,
+																	is_final: true,
+																},
+															},
+															{
+																ipor: {
+																	raw_version: release,
+																	is_final: false,
```

Comment:
We definitely could use a comment here to explain why we query those two different fields on each case.